lib/gmp/native_gmp.nit

```diff
@@ -0,0 +1,266 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Low-level GMP features
+module native_gmp is ldflags("-lgmp")
+
+in "C header" `{
+    #include <gmp.h>
+`}
+
+# Multy precision Integer
+extern class NativeLong `{mpz_ptr`}
+
+    # Initializing
+
+    new `{
+        mpz_ptr self =  (mpz_ptr)malloc(sizeof(mpz_t));
+        mpz_init(self);
+        return self;
+    `}
+
+    # Arithmetic Functions
+
+    fun add(res: NativeLong, op: NativeLong) `{
+        mpz_add(res, self, op);
+    `}
+
+    fun add_uint(res: NativeLong, op: UInt64) `{
+        mpz_add_ui(res, self, *op);
+    `}
+
+    fun sub(res: NativeLong, op: NativeLong) `{
+        mpz_sub(res, self, op);
+    `}
+
+    fun sub_uint(res: NativeLong, op: UInt64) `{
+        mpz_sub_ui(res, self, *op);
+    `}
+
+    fun mul(res: NativeLong, op: NativeLong) `{
+        mpz_mul(res, self, op);
+    `}
+
+    fun mul_int(res: NativeLong, op: Int) `{
+        mpz_mul_si(res, self, op);
+    `}
+
+    fun neg(res: NativeLong) `{
+        mpz_neg(res, self);
+    `}
+
+    fun abs(res: NativeLong) `{
+        mpz_abs(res, self);
+    `}
+
+    fun div(res: NativeLong , op: NativeLong) `{
+        mpz_tdiv_q(res, self, op);
+    `}
+
+    fun div_uint(res: NativeLong , op: UInt64) `{
+        mpz_tdiv_q_ui(res, self, *op);
+    `}
+
+    fun mod(res: NativeLong , op: NativeLong) `{
+        mpz_mod(res, self, op);
+    `}
+
+    fun mod_uint(res: NativeLong , op: UInt64) `{
+        mpz_mod_ui(res, self, *op);
+    `}
+
+    fun pow_uint(res: NativeLong , op: UInt64) `{
+        mpz_pow_ui(res, self, *op);
+    `}
+
+    #Number Theoretic Functions
+
+    fun probab_prime(reps: Int32): Int32 `{
+        return mpz_probab_prime_p(self, reps);
+    `}
+
+    fun next_prime(res: NativeLong) `{
+        mpz_nextprime(res, self);
+    `}
+
+    fun gcd(res: NativeLong, op: NativeLong) `{
+        mpz_gcd(res, self, op);
+    `}
+
+    fun gcd_uint(res: NativeLong , op: UInt64) `{
+        mpz_gcd_ui(res, self, *op);
+    `}
+
+    # Comparison Functions
+
+    fun cmp(op: NativeLong): Int32 `{
```

Comment:
For comparisons, I'd suggest returning an Int since their use is a bit less tedious than the Intxx variants (no need for a suffix when dealing with literal values).